src/Components/AdvancedSearch/Internal/QueryBuilder/QueryBuilderRow.tsx

```diff
@@ -0,0 +1,307 @@
+
+    // Update operator options
+    useEffect(() => {
+        if (selectedProperty) {
+            const operatorOptions = getOperators(selectedProperty.data.type);
+            setOperatorOptions(operatorOptions);
+            if (operatorOptions.length) {
+                setSelectedOperator(operatorOptions[0]);
+            }
+        }
+    }, [selectedProperty]);
+
+    // Update snippet on any input change
+    useEffect(() => {
+        if (selectedProperty) {
+            onUpdateSnippet(rowId, {
+                combinator: selectedCombinator
+                    ? selectedCombinator
+                    : DEFAULT_COMBINATOR,
+                operatorData: selectedOperator
+                    ? selectedOperator.data
+                    : DEFAULT_OPERATOR,
+                property: selectedProperty.data.name,
+                value: selectedValue
+                    ? selectedValue
+                    : getDefaultPropertyValues(selectedProperty.data.type)
+            });
+        }
+    }, [selectedCombinator, selectedOperator, selectedProperty, selectedValue]);
+
+    const onChangeOperator = (
+        _event: React.FormEvent<HTMLDivElement>,
+        option?: IDropdownOption<OperatorData>,
+        _index?: number
+    ) => {
+        setSelectedOperator(option);
+    };
+
+    const onChangeCombinator = (
+        _event: React.FormEvent<HTMLDivElement>,
+        option?: IDropdownOption,
+        _index?: number
+    ) => {
+        setSelectedCombinator(option.text);
+    };
+
+    const onChangePropertySelected = (
+        newValue: PropertyOption,
+        _actionMeta: SelectOptionActionMeta<PropertyOption>
+    ) => {
+        setSelectedProperty(newValue);
+        onChangeProperty(rowId, newValue.data?.name, newValue.data?.type);
+    };
+
+    const onChangeValueField = (
+        _event: React.SyntheticEvent<HTMLElement, Event>,
+        newValue?: string
+    ) => {
+        setSelectedValue(newValue);
+        onChangeValue(rowId, newValue);
+    };
+
+    const onChangeDropdownValue = (
+        _event: React.FormEvent<HTMLDivElement>,
+        option?: IDropdownOption<any>,
+        _index?: number
+    ) => {
+        setSelectedValue(option.text);
+        onChangeValue(rowId, option.text);
+    };
+
+    const propertySelectorStyles = reactSelectStyles(iTheme, isRemoveDisabled);
+
+    const Group = (props) => <components.Group {...props} />;
+
+    const Menu = (props) => (
+        <Callout
+            target={`#${propertySelectorId}`}
+            styles={classNames.subComponentStyles.propertyCallout}
+            isBeakVisible={false}
+        >
+            <BaseComponent theme={theme} disableDefaultStyles={true}>
+                <components.MenuList
+                    {...props}
+                    styles={propertySelectorStyles.menuList}
+                />
+            </BaseComponent>
+        </Callout>
+    );
+
+    const renderValueField = () => {
```

Comment:
Can we prepend the type of the property with icons? We did it in the past for `ValueWidgetBuilder` using `DTDLPropertyIconographyMap`
<img width="328" alt="image" src="https://user-images.githubusercontent.com/45217314/188199539-f91ae015-d1db-4668-a535-2bd0cd62ceb5.png">

![image](https://user-images.githubusercontent.com/45217314/188199997-9383f3c4-7678-4188-969a-16b262da80e2.png)

Reply:
I'll fix this in a following PR, thanks for the reminder.